plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -1,25 +1,89 @@
 import NotificationCenter
 
+enum InAppConsentStrategy: String {
+    case autoEnroll = "auto-enroll"
+    case explicitByUser = "explicit-by-user"
+    case disabled = "in-app-disabled"
+}
+
 @objc(Optimove_Cordova) class OptimoveSDKPlugin : CDVPlugin {
     private static var optimovePluginInstance: OptimoveSDKPlugin!
+    private static var cordovaCommand: CDVInvokedUrlCommand? = nil
 
     private static let optimoveCredentialsKey = "optimoveCredentials"
     private static let optimoveMobileCredentialsKey = "optimoveMobileCredentials"
+    private static let inAppConsentStrategy = "optimoveInAppConsentStrategy"
+    //private static let enableDeferredDeepLinking = "optimoveEnableDeferredDeepLinking"
+
+    private static var pendingPush: PushNotification? = nil
+
+    // ========================== INITIALIZATION ==========================
+
+    override func pluginInitialize() {
+        OptimoveSDKPlugin.optimovePluginInstance = self
+    }
 
-    private static var config: OptimoveConfig? = {
+    @objc(didFinishLaunching:)
+    static func didFinishLaunching(notification: Notification) {
         let configPath = Bundle.main.path(forResource: "optimove", ofType: "plist")
 
         guard let configPath = configPath else {
             print("optimove.plist NOT FOUND")
-            return nil
+            return
         }
 
         guard let configValues: [String: String] = NSDictionary(contentsOfFile: configPath) as? [String: String] else {
             print("optimove.plist IS NOT VALID")
-            return nil
+            return
+        }
+
+        guard let builder = getConfigBuilder(configValues: configValues) else{
+            return
+        };
+
+        builder.setPushOpenedHandler(pushOpenedHandlerBlock: { notification in
+            guard let pluginInstance = OptimoveSDKPlugin.optimovePluginInstance, let _ = OptimoveSDKPlugin.cordovaCommand else {
+                pendingPush = notification
+
+                return
+            }
+
+            pluginInstance.sendMessageToJs(type: "pushOpened", data: getPushNotificationMap(pushNotification: notification))
+        })
+
+        if #available(iOS 10, *) {
+            builder.setPushReceivedInForegroundHandler(pushReceivedInForegroundHandlerBlock: { notification, completionHanlder in
+                if (OptimoveSDKPlugin.optimovePluginInstance != nil){
+                    OptimoveSDKPlugin.optimovePluginInstance.sendMessageToJs(type: "pushReceived", data: getPushNotificationMap(pushNotification: notification))
+                }
+
+                completionHanlder(UNNotificationPresentationOptions.alert)
+            })
+        }
+
+        if (configValues[inAppConsentStrategy] != InAppConsentStrategy.disabled.rawValue) {
+            builder.enableInAppMessaging(inAppConsentStrategy: configValues[inAppConsentStrategy] == InAppConsentStrategy.autoEnroll.rawValue ? OptimoveSDK.InAppConsentStrategy.autoEnroll : OptimoveSDK.InAppConsentStrategy.explicitByUser)
```

Comment:
Prefer explicit coverage of cases / no 'default' else? (maybe acceptable as unlikely changing options too soon)

Reply:
prob was copied from Flutter, changed to switch